Unity-Technologies/ScriptableRenderPipeline
# Trio of PRs from external customer plus updates

This PR is a superset of three PRs submited by soraryu, an external customer, plus a few updates of my own. 

I made dev branch xr_sg_urp_dennisd from SRP master then merged in three dev branches from prefrontalcortex remote per the customer's three PRs:
PR #4802: URP/Shadergraph/XR GPU Instancing Tests
PR #4803: Added Test for sampling OpaqueTexture in XR through ShaderGraph 
PR #5022: Workaround for Single-Pass and Single-Pass Instanced testing issues in #4802

Since then I have merged more recent master commits into my branch, toggled the stereo mode from Single Pass (double-wide) to Stereo Instancing for ShaderGraphUniversalStereo, and disabled 003_xr_OpaqueTexture due to case 1193371. 

After soraryu's three PRs are approved by relevant reviewers, this PR could be merged to include my small changes.  Alternatively, after soraryu's three PRs are approved, this PR could be merged instead since it includes soraryu's changes plus mine plus more recent master merges.  Not sure which approach is best from a code management perspective... 


**DONT FORGET TO ADD A CHANGELOG**

### Checklist for PR maker
- Have you added a Label? : HDRP, Universal, ShaderGraph etc...
Yes. 

- Have you added a label for backport (if needed)? : need-backport-2019.3  .  When the PR is backported the label will be change ton backported-2019.3
No. 

- Have you added a changelog? Each package have a changelog.
No. 

- Have you updated or added the documentation for you PR? When property name is changed, when a feature behavior is change, when adding a new features, think to update the documentation in the same PR.
N/A; this PR only adds tests. 

- Have you added a graphic test for your PR (if needed)? When adding new feature or discovering a bug that isn't cover by a test, please add a graphic test
This PR only adds tests. 

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status
I examined the new tests in the Editor, skimmed through the code that works around Stereo Instancing screen capture bugs (case ), and ran "fast trunk" ABV on Yamato (see below).  

**Manual Tests**: What did you do?
- [YES ] Opened test project + Run graphic tests locally
- [YES (via Yamato) ] Built a player
- [ N/A ] Checked new UI names with UX convention
- [ N/A ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ NO  ] C# and shader warnings (supress shader cache to see them)
- [ NO ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
The goal of this PR is to improve our test automation. 

**Yamato**: (Select your branch):
"_ABV for SRP repository - fast-trunk" in green for the latest commit on my dev branch except for three pipelines which became green after several re-runs.  Here are the links that add up to green ABV: 

https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/xr_sg_urp_dennisd/.yamato%252Fupm-ci-abv.yml%2523all_project_ci_fast-trunk/805898/job/pipeline

https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/xr_sg_urp_dennisd/.yamato%252Fupm-ci-hdrp.yml%2523HDRP_Win_DX11_playmode_fast-trunk/806446/job

https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/xr_sg_urp_dennisd/.yamato%252Fupm-ci-hdrp.yml%2523HDRP_Win_DX11_playmode%2520XR_fast-trunk/806549/job

https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/xr_sg_urp_dennisd/.yamato%252Fupm-ci-universal_stereo.yml%2523Universal_Stereo_Windows64_Standalone_fast-trunk/806454/job

Any test projects to go with this to help reviewers?
This PR is entirely test projects and the corresponding assets, settings changes, reference images, etc. plus a work around to enable Stereo Instancing screen captures.  

---
### Comments to reviewers
Notes for the reviewers you have assigned.

I'll be on-line occasionally during the extended USA holiday weekend if any clarifications are needed.  Feel free to block this PR if you see a major problem that I overlooked.  However, if you see minor issues then I would prefer to fix them in a follow-up PR.  I plan to submit PRs for new SRP XR tests on a cadence of every other week for the near future.  Thanks!
